addons/purchase_requisition/models/purchase.py

```diff
@@ -109,6 +109,18 @@ def write(self, vals):
 class PurchaseOrderLine(models.Model):
     _inherit = 'purchase.order.line'
 
+    @api.depends('product_id', 'date_order')
+    def _compute_account_analytic_id(self):
+        for rec in self:
+            if not rec.order_id.requisition_id:
```

Comment:
Wasn't the default behavior before https://github.com/odoo/odoo/commit/eadd0be5bb119e5af2c1c721f8de84502c1844b0 to use the default analytic account if not is specified by the purchase requisition and one exists for that product? (I'm not sure, should maybe be double checked with a PO)

This solution will make it so if the requisition line for the product has no analytic account assigned then it won't use the default one either and stay false :thinking: Also perhaps a test will help make it clear what the final flow should be :wink:

Reply:
Hello @ticodoo 
Thank you for the review.
I asked the PO, it seems that the behavior before: https://github.com/odoo/odoo/commit/eadd0be5bb119e5af2c1c721f8de84502c1844b0 was wrong. So if no analytical account is specified in the purchase requisition line, the one defined by default for the product should not be used when creating a PO. however when a new requisition line is added, the product's default analytic account and analytic tag must be computed